tests/test_resolutions.py

```diff
@@ -570,12 +572,35 @@ def test_the_test(self) -> None:
         assert m.matches("Spam spam spam spam baked beans and spam")
         assert not m.matches("What do you mean Eugh?!")
 
+    def test_the_test_with_compile(self) -> None:
+        m = Matches(re.compile(r"([Ss]pam ?)+")).resolve()
+
+        assert m.matches("Spam spam spam spam baked beans and spam")
+        assert not m.matches("What do you mean Eugh?!")
+
     def test_description(self) -> None:
         test_match = r"(spam)+"
 
         m = Matches(test_match)
 
-        expected_description = 'Text matching the pattern r"(spam)+".'
+        expected_description = "Text matching the pattern r'(spam)+'."
+        assert m.describe() == expected_description
+
+    def test_beat_logging(self, caplog: pytest.LogCaptureFixture) -> None:
+        caplog.set_level(logging.INFO)
+        Matches(r"\tpattern").resolve()
+
+        assert [r.msg for r in caplog.records] == [
+            r"... hoping it's text matching the pattern r'\tpattern'.",
+            r"    => a string matching '\tpattern'",
+        ]
+
+    def test_description_with_compile(self) -> None:
+        test_match = re.compile(r"(spam)+")
+
+        m = Matches(test_match)
+
+        expected_description = "Text matching the pattern r'(spam)+'."
```

Comment:
Same goes for this test; we've just grouped all the "describe correctly" tests in one in other ones (though currently the only examples i can think of are for testing `SeeAllOf` and `SeeAnyOf`).